File: CS/RestrictedDraggingExample/Docking/CustomDockingHandlerStateVS2005.cs
using System;
using System.Drawing;
using DevExpress.XtraBars.Docking.Helpers;

namespace RestrictedDraggingExample.Docking {
    public class CustomDockingHandlerStateVS2005 :DockingHandlerStateVS2005 {
        public CustomDockingHandlerStateVS2005(DockPanelMouseHandler handler): base(handler) { }

        protected override Point CalcDockPanelLocationWithOffset(Point point) {
            Point result = base.CalcDockPanelLocationWithOffset(point);
            Point topLeft = DockManager.Form.PointToScreen(DockManager.Form.ClientRectangle.Location);
            Point bottomright = DockManager.Form.PointToScreen(new Point(DockManager.Form.ClientRectangle.Right,
                DockManager.Form.ClientRectangle.Bottom));
            return new Point(Math.Max(Math.Min(result.X, bottomright.X - Owner.Width),
                topLeft.X), Math.Max(Math.Min(result.Y, bottomright.Y - Owner.Height),
                topLeft.Y));
        }
    }
}

File: VB/RestrictedDraggingExample/Docking/CustomDockingHandlerStateVS2005.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.XtraBars.Docking.Helpers

Namespace RestrictedDraggingExample.Docking
	Public Class CustomDockingHandlerStateVS2005
		Inherits DockingHandlerStateVS2005
		Public Sub New(ByVal handler As DockPanelMouseHandler)
			MyBase.New(handler)
		End Sub

		Protected Overrides Function CalcDockPanelLocationWithOffset(ByVal point As Point) As Point
			Dim result As Point = MyBase.CalcDockPanelLocationWithOffset(point)
			Dim topLeft As Point = DockManager.Form.PointToScreen(DockManager.Form.ClientRectangle.Location)
			Dim bottomright As Point = DockManager.Form.PointToScreen(New Point(DockManager.Form.ClientRectangle.Right, DockManager.Form.ClientRectangle.Bottom))
			Return New Point(Math.Max(Math.Min(result.X, bottomright.X - Owner.Width), topLeft.X), Math.Max(Math.Min(result.Y, bottomright.Y - Owner.Height), topLeft.Y))
		End Function
	End Class
End Namespace
